### ISSUE TITLE
The frontgate/drone image version should match openpitrix scheduler image version

### ISSUE THREAD

**Issue by chilianyi** [openpitrix/openpitrix#599]
The frontgate/drone services are started in vmbased runtime's instances, use latest image now. 
When openpitrix scheduler modules are upgrading, may add new rpc request from pilot to frontgate, while frontgate uses old image, which has not the new rpc request handler function. 

The problem is: how to upgrade frontgate's image to match openpitrix scheduler image version

**Comment by Hevienz** [comment]
I found a solution (https://github.com/v2tec/watchtower), and I think we can try it.

It will watching on the container that you cared about, if it's image has changed, it will pull the new image, and restart the container with the same option.

**Comment by chilianyi** [comment]
Nice, the solution is good for the situation when use latest image. 

But I have another question, if a private openpitrix, use openpitrix v0.2.0, when our frontgate image upgrade to v0.2.4, his/her openpitrix upgrade to v0.2.4, how do his/her frontgate upgrade to v0.2.4.

**Comment by chilianyi** [comment]
Pilot, Frontgate, Drone should keep the same version. 
When pilot upgrade, frontgate and drone should upgrade too, so frontgate and drone should crontab check if pilot's version is upgraded. 

Now need developer put `drone` into his/her image, drone should also keep the same version with frontgate and pilot. If the sidecar pattern is useful here? Divide into two image, share the file system.

**Comment by Hevienz** [comment]
How about this, one container check frontgate or drone has an upgrade binary relaese periodicity, if yes, download it into `/opt/openpitrix/bin`, and this directory is mounted from `/opt/openpitrix/bin`, just like below:

```bash
docker run -v /opt/openpitrix/bin:/opt/openpitrix/bin updater
```

Another container mounted the same directory, he will run frontgate or drone from that directory.

```bash
docker run -v /opt/openpitrix/bin:/opt/openpitrix/bin metadata /opt/openpitrix/bin/frontgate
# or
docker run -v /opt/openpitrix/bin:/opt/openpitrix/bin metadata /opt/openpitrix/bin/drone
```

**Comment by Hevienz** [comment]
Optimized solution:

1 Pilot serves a rpc request which return its version.

2 Frontgate and drone check the version of pilot through the rpc request periodically. If they found their version not match with the version of pilot. They download new binary release from some url respectively, then they exit.

3 There must be a frontgate or drone exist in their image, maybe located in /usr/local/bin/, and new binary release will be located in /opt/openpitrix/bin. When container is starting, he will check if there is a new release, if so, start the new one, otherwise the origin one.

**Comment by Hevienz** [comment]
Optimized solution draft 2:

1 Pilot serves a rpc request which return its version.

2 Frontgate and drone check the version of pilot through the rpc request periodically. If they found their version not match with the version of pilot, then they write the version of pilot to /opt/openpitrix/conf/pilot-version and exit.

3 Frontgate and drone are started by runit, the script used by runit will check if their version is matched with /opt/openpitrix/conf/pilot-version, if not, the script will try to fetch the matched version, and start with the new version.

**Comment by Hevienz** [comment]
Merged #624.
